src/devices/Mfrc522/Mfrc522Controller.cs

```diff
@@ -0,0 +1,447 @@
+            {
+                WriteSpi(FIFODataReg, sendData[i]);
+                i++;
+            }
+
+            WriteSpi(CommandReg, (byte)command);
+
+            if (command == Transceive)
+            {
+                SetBitMask(BitFramingReg, 0x80);
+            }
+
+            i = 2000;
+            do
+            {
+                n = ReadSpi((byte)CommIrqReg);
+            }
+            while (--i != 0 && (n & 0x01) == 0 && (n & waitIRq) == 0);
+
+            ClearBitMask(BitFramingReg, 0x80);
+
+            if (i == 0) return (status, backData.ToArray(), backLen);
+
+            if ((ReadSpi(ErrorReg) & 0x1B) == 0x00)
+            {
+                status = OK;
+
+                if (Convert.ToBoolean(n & irqEn & 0x01))
+                {
+                    status = NoTag;
+                }
+
+                if (command == Transceive)
+                {
+                    n = ReadSpi(FIFOLevelReg);
+                    byte? lastBits = (byte)(ReadSpi(ControlReg) & 0x07);
+                    if (lastBits != 0)
+                    {
+                        backLen = (byte)(((n - 1) * 8) + (byte)lastBits);
+                    }
+                    else
+                    {
+                        backLen = (byte)(n * 8);
+                    }
+
+                    if (n == 0)
+                    {
+                        n = 1;
+                    }
+
+                    if (n > MAX_LEN)
+                    {
+                        n = MAX_LEN;
+                    }
+
+                    i = 0;
+                    while (i < n)
+                    {
+                        backData.Add(ReadSpi(FIFODataReg));
+                        i++;
+                    }
+                }
+            }
+            else
+            {
+                status = Error;
+            }
+
+            return (status, backData.ToArray(), backLen);
+        }
+
+        public (Status status, int backBits) Request(RequestMode requestMode)
+        {
+            var tagType = new List<byte> { (byte)requestMode };
+
+            WriteSpi(BitFramingReg, 0x07);
+
+            var (status, backData, backBits) = SendCommand(Transceive, tagType.ToArray());
+
+            if ((status != OK) | (backBits != 0x10))
+            {
+                status = Error;
+            }
+
+            return (status, backBits);
+        }
+
+        public (Status status, byte[] data) AntiCollision()
+        {
+            byte serNumCheck = 0;
+
+            var serNum = new List<byte>();
+
+            WriteSpi(BitFramingReg, (byte)0x00);
+
+            serNum.Add((byte)RequestMode.AntiCollision);
+            serNum.Add(0x20);
+
+            var (status, backData, _) = SendCommand(Transceive, serNum.ToArray());
+
+            var i = 0;
+            if (status == OK)
+            {
+                i = 0;
+            }
+
+            if (backData.Length == 5)
+            {
+                while (i < 4)
```

Comment:
I presume this is some kind of CRC calculation? if so please create a function from that